src/lib-zephyr/alloc.c

```diff
@@ -0,0 +1,280 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ *
+ */
+
+#include <sof/init.h>
+#include <rtos/alloc.h>
+#include <sof/drivers/idc.h>
+#include <rtos/interrupt.h>
+#include <sof/drivers/interrupt-map.h>
+#include <sof/lib/dma.h>
+#include <sof/schedule/schedule.h>
+#include <platform/drivers/interrupt.h>
+#include <platform/lib/memory.h>
+#include <sof/platform.h>
+#include <sof/lib/notifier.h>
+#include <sof/lib/pm_runtime.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/component_ext.h>
+#include <sof/trace/trace.h>
+#include <rtos/wait.h>
+
+/* Zephyr includes */
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/pm/policy.h>
+#include <version.h>
+#include <zephyr/sys/__assert.h>
+#include <soc.h>
+
+#if !CONFIG_KERNEL_COHERENCE
+#include <zephyr/arch/xtensa/cache.h>
+#endif
+
+#if CONFIG_SYS_HEAP_RUNTIME_STATS
+#include <zephyr/sys/sys_heap.h>
+#endif
+
+LOG_MODULE_REGISTER(mem_allocator, CONFIG_SOF_LOG_LEVEL);
+
+/*
+ * Memory - Create Zephyr HEAP for SOF.
+ *
+ * Currently functional but some items still WIP.
+ */
+
+#ifndef HEAP_RUNTIME_SIZE
+#define HEAP_RUNTIME_SIZE	0
```

Comment:
looks like this is a xtos only option ? I wonder where we are still using in Zephyr config ?

Reply:
IMX platform only - to be removed